jetty-core/jetty-websocket/jetty-websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/internal/ServerUpgradeResponseImpl.java

```diff
@@ -55,6 +56,8 @@ public String getAcceptedSubProtocol()
     public void setAcceptedSubProtocol(String protocol)
     {
         negotiation.setSubprotocol(protocol);
+        // Use the nested headers to avoid recursion.
+        response.getHeaders().put(HttpHeader.SEC_WEBSOCKET_SUBPROTOCOL, protocol);
```

Comment:
It is not clear why this was moved from `WebScoketNegotiation.setProtocol`?
The comment is appreciated, but would be good to say who checks it and thus avoids recursing?

Reply:
The comment was expanded to specify that the header was set on the _nested_ headers, not the wrapped ones.